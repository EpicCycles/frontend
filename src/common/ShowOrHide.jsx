import React from 'react';
import { Icon } from 'semantic-ui-react';
import * as PropTypes from 'prop-types';

const ShowOrHide = props => {
  const { componentKey, isShown, detailKey, hideDetail, showDetail } = props;
  return (
    <Icon
      key={componentKey}
      id={componentKey}
      name={`toggle ${isShown ? 'right' : 'down'}`}
      onClick={() => (!!isShown ? hideDetail(detailKey) : showDetail(detailKey))}
    />
  );
};
ShowOrHide.propTypes = {
  componentKey: PropTypes.any.isRequired,
  isShown: PropTypes.bool,
  hideDetail: PropTypes.func.isRequired,
  showDetail: PropTypes.func.isRequired,
};
export default ShowOrHide;
