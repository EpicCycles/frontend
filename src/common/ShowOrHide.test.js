import ShowOrHide from './ShowOrHide';

describe('ShowOrHide', () => {
  it('it renders without failing', () => {
    const hideDetail = jest.fn();
    const showDetail = jest.fn();
    const component = shallow(
      <ShowOrHide
        componentKey="mykey"
        isShown={false}
        hideDetail={hideDetail}
        showDetail={showDetail}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it('it calls hide when details shown and icon clicked', () => {
    const hideDetail = jest.fn();
    const showDetail = jest.fn();
    const component = shallow(
      <ShowOrHide
        componentKey="mykey"
        isShown={false}
        hideDetail={hideDetail}
        showDetail={showDetail}
      />,
    );

    expect(component.find('#mykey')).toHaveLength(1);
    component.find('#mykey').simulate('click');
    expect(showDetail).toBeCalled();
  });
});
