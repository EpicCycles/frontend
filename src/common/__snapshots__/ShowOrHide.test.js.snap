// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowOrHide it renders without failing 1`] = `
<Icon
  as="i"
  id="mykey"
  key="mykey"
  name="toggle down"
  onClick={[Function]}
/>
`;
